File: app/src/main/java/com/icerrate/popularmovies/view/movies/search/SearchMoviesContract.java
package com.icerrate.popularmovies.view.movies.search;

import com.icerrate.popularmovies.data.model.Movie;
import com.icerrate.popularmovies.data.model.PaginatedResponse;
import com.icerrate.popularmovies.data.model.Review;
import com.icerrate.popularmovies.data.model.Trailer;
import com.icerrate.popularmovies.view.common.BaseView;

import java.util.ArrayList;
import java.util.List;

/**
 * @author Ivan Cerrate.
 */

public interface SearchMoviesContract {

    interface View {

        void showProgressBar(boolean show);

        void showError(String errorMessage);

        void resetMovies();

        void showMovies(List<Movie> movies);

        void showSearchHint(boolean show);

        void showNoDataView(boolean show);

        void showFooterProgress(boolean show);

        void goToMovieDetail(Movie movie);
    }

    interface Presenter {

        void searchMovies(String query);

        void loadNextMoviesPage();

        void loadMovies();

        void onMovieItemClick(Movie movie);

        String getQuery();

        PaginatedResponse<Movie> getMoviesPaginatedResponse();

        void loadPresenterState(String query, PaginatedResponse<Movie> moviePaginatedResponse);
    }
}

File: app/src/main/java/com/icerrate/popularmovies/view/movies/search/SearchMoviesContract.kt
package com.icerrate.popularmovies.view.movies.search

import com.icerrate.popularmovies.data.model.Movie
import com.icerrate.popularmovies.data.model.PaginatedResponse

/**
 * @author Ivan Cerrate.
 */
interface SearchMoviesContract {

    interface View {

        fun showProgressBar(show: Boolean)

        fun showError(errorMessage: String)

        fun resetMovies()

        fun showMovies(movies: List<Movie>)

        fun showSearchHint(show: Boolean)

        fun showNoDataView(show: Boolean)

        fun showFooterProgress(show: Boolean)

        fun goToMovieDetail(movie: Movie)
    }

    interface Presenter {

        fun searchMovies(query: String)

        fun loadNextMoviesPage()

        fun loadMovies()

        fun onMovieItemClick(movie: Movie)

        fun getQuery(): String?

        fun getMoviesPaginatedResponse(): PaginatedResponse<Movie>?

        fun loadPresenterState(query: String?, moviePaginatedResponse: PaginatedResponse<Movie>?)
    }
}